src/effects/audiounits/AudioUnitEffect.cpp

```diff
@@ -973,8 +959,10 @@ bool AudioUnitEffect::SupportsAutomation() const
 
 bool AudioUnitEffect::CreateAudioUnit()
 {
-   auto result = AudioComponentInstanceNew(mComponent, &mUnit);
-   return (!result && mUnit != nullptr);
+   AudioUnit unit{};
+   auto result = AudioComponentInstanceNew(mComponent, &unit);
+   mUnit.reset(unit);
```

Comment:
I wonder if it would make sense to check `result ` before resetting the smart pointer?

Reply:
I will do that when I merge it!